src/bbd/geocoder/geocoder.py

```diff
@@ -0,0 +1,582 @@
+            str and setting defaults as necessary. (see defaults)
+        For dataframe: 
+            data should either contain 1 column
+            OR 
+            columns using follow VAN default column names (not case sensitive):
+             "Address" - column of str street component of address
+             "City" - column of str city component of address
+             "County" - column of str county component of address
+             "State" - column of str state component of address
+             "Zip5" - column of str 5-digit postal code for address
+             "Country" - column of str country component of address
+                         (if applicable, default will set country to US)
+        For list of dicts:
+            Each item should be a dict of the form
+             "Address" : str street component of address
+             "City" : str city component of address
+             "County" : str county component of address
+             "State" : str state component of address
+             "Zip5" : str 5-digit postal code for address
+             "Country" : str country component address
+                         (if applicable, default will set country to US)
+        For list of strings:
+            Each item should str full address to pass as query.
+
+    email : str
+        A valid email address as str to pass to Nominatim as user_agent.
+        See Nominatim's Usage Policy.
+
+    path : str or pathlib Path object
+        A path to where geocoded data will be saved.
+
+    batch_size : (Optional) int
+        Number of batches to geocode at a time. Useful for running subset of
+        data. Will determin size of batch for run(num_batches) and size of 
+        progress bar while running.
+
+    defaults : (Optional) dict
+        A dictionary of default values for query. Should be of form:
+        {
+            "Address" : address_default,
+            "City" : city_default,
+            "County" : county_default,
+            "State" : state_default,
+            "Zip5" : postal_default,
+            "Country" : country_default
+        }
+        Will set country to United States by default.
+
+    keep_index : (Optional) bool
+        Whether to use the index of data for the geocoded result. If 
+        False, will use 
+        It is recommended to use the same index for the inputted data
+        to easily merge the data.
+        True by default.
+
+    Properties
+    ----------
+    street_encode : func
+        A function of 1 argument used to encode the street address. The 
+        default is street_encode(address) used to remove secondary 
+        descriptors from the street address. This is recommended 
+        because many addresses in OpenStreetMaps do not have
+        secondary descriptors attached and therefore will not be 
+        geocoded.
+
+    geocoder : str
+        Enter email as str to change geocoder user_agent.
```

Comment:
So I had just found properties and went a little ham when I was playing around with them... resulting in this kinda weird unituitive thing. 

Basically I wanted geocoder - which is basically a `geopy.geocoders.Nominatim` instance fed into a `geopy.extras.RateLimiter` with that string value set as the user_agent. I wanted users to be able to pass their own `geopy.geocoder` in a  `geopy.extras.RateLimiter`or an email as a str to change the geocoder but I forgot to finish updating the documentation. It might be helpful to have a more descriptive name... 

I was thinking the use-case for that would be if you wanted to run 1 batch, then change emails (b/c maybe Nominatim started complaining or smth) then keep going.

Reply:
Yeah, the only problem is functionally to have the email change do anything is also changing the geocoder. What would a better name for email be that makes it clear it will mutate the underlying geocoder. 

Also I'm changing it so it only accepts a geopy.geocoder that is wrapped in a geopy.extras.RateLimiter b/c otherwise the geocoder will throw up a bunch of errors or possibly result in the service banning the user's domain.